// devextreme-vue@24.2.10/cjs/responsive-box.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxRow = exports.DxLocation = exports.DxItem = exports.DxCol = exports.DxResponsiveBox = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const responsive_box_1 = __importDefault(require("devextreme/ui/responsive_box"));
const index_2 = require("./core/index");
const componentConfig = {
    props: {
        cols: Array,
        dataSource: [Array, Object, String],
        disabled: Boolean,
        elementAttr: Object,
        height: [Function, Number, String],
        hoverStateEnabled: Boolean,
        itemHoldTimeout: Number,
        items: Array,
        itemTemplate: {},
        onContentReady: Function,
        onDisposing: Function,
        onInitialized: Function,
        onItemClick: Function,
        onItemContextMenu: Function,
        onItemHold: Function,
        onItemRendered: Function,
        onOptionChanged: Function,
        rows: Array,
        rtlEnabled: Boolean,
        screenByWidth: Function,
        singleColumnScreen: String,
        visible: Boolean,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:cols": null,
        "update:dataSource": null,
        "update:disabled": null,
        "update:elementAttr": null,
        "update:height": null,
        "update:hoverStateEnabled": null,
        "update:itemHoldTimeout": null,
        "update:items": null,
        "update:itemTemplate": null,
        "update:onContentReady": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onItemClick": null,
        "update:onItemContextMenu": null,
        "update:onItemHold": null,
        "update:onItemRendered": null,
        "update:onOptionChanged": null,
        "update:rows": null,
        "update:rtlEnabled": null,
        "update:screenByWidth": null,
        "update:singleColumnScreen": null,
        "update:visible": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = responsive_box_1.default;
        this.$_hasAsyncTemplate = true;
        this.$_expectedChildren = {
            col: { isCollectionItem: true, optionName: "cols" },
            item: { isCollectionItem: true, optionName: "items" },
            row: { isCollectionItem: true, optionName: "rows" }
        };
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxResponsiveBox = (0, vue_1.defineComponent)(componentConfig);
exports.DxResponsiveBox = DxResponsiveBox;
const DxColConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:baseSize": null,
        "update:ratio": null,
        "update:screen": null,
        "update:shrink": null,
    },
    props: {
        baseSize: [Number, String],
        ratio: Number,
        screen: String,
        shrink: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxColConfig);
const DxCol = (0, vue_1.defineComponent)(DxColConfig);
exports.DxCol = DxCol;
DxCol.$_optionName = "cols";
DxCol.$_isCollectionItem = true;
const DxItemConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:disabled": null,
        "update:html": null,
        "update:location": null,
        "update:template": null,
        "update:text": null,
        "update:visible": null,
    },
    props: {
        disabled: Boolean,
        html: String,
        location: [Array, Object],
        template: {},
        text: String,
        visible: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxItemConfig);
const DxItem = (0, vue_1.defineComponent)(DxItemConfig);
exports.DxItem = DxItem;
DxItem.$_optionName = "items";
DxItem.$_isCollectionItem = true;
DxItem.$_expectedChildren = {
    location: { isCollectionItem: true, optionName: "location" }
};
const DxLocationConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:col": null,
        "update:colspan": null,
        "update:row": null,
        "update:rowspan": null,
        "update:screen": null,
    },
    props: {
        col: Number,
        colspan: Number,
        row: Number,
        rowspan: Number,
        screen: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxLocationConfig);
const DxLocation = (0, vue_1.defineComponent)(DxLocationConfig);
exports.DxLocation = DxLocation;
DxLocation.$_optionName = "location";
DxLocation.$_isCollectionItem = true;
const DxRowConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:baseSize": null,
        "update:ratio": null,
        "update:screen": null,
        "update:shrink": null,
    },
    props: {
        baseSize: [Number, String],
        ratio: Number,
        screen: String,
        shrink: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxRowConfig);
const DxRow = (0, vue_1.defineComponent)(DxRowConfig);
exports.DxRow = DxRow;
DxRow.$_optionName = "rows";
DxRow.$_isCollectionItem = true;
exports.default = DxResponsiveBox;
