// devextreme-vue/esm/speed-dial-action.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

import { defineComponent } from "vue";
import { prepareComponentConfig } from "./core/index";
import SpeedDialAction from "devextreme/ui/speed_dial_action";
const componentConfig = {
    props: {
        accessKey: String,
        activeStateEnabled: Boolean,
        elementAttr: Object,
        focusStateEnabled: Boolean,
        hint: String,
        hoverStateEnabled: Boolean,
        icon: String,
        index: Number,
        label: String,
        onClick: Function,
        onContentReady: Function,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        rtlEnabled: Boolean,
        tabIndex: Number,
        visible: Boolean
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:accessKey": null,
        "update:activeStateEnabled": null,
        "update:elementAttr": null,
        "update:focusStateEnabled": null,
        "update:hint": null,
        "update:hoverStateEnabled": null,
        "update:icon": null,
        "update:index": null,
        "update:label": null,
        "update:onClick": null,
        "update:onContentReady": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:rtlEnabled": null,
        "update:tabIndex": null,
        "update:visible": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = SpeedDialAction;
        this.$_hasAsyncTemplate = true;
    }
};
prepareComponentConfig(componentConfig);
const DxSpeedDialAction = defineComponent(componentConfig);
export default DxSpeedDialAction;
export { DxSpeedDialAction };
