// devextreme-vue/cjs/chat.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxUser = exports.DxTypingUser = exports.DxMessageTimestampFormat = exports.DxItem = exports.DxDayHeaderFormat = exports.DxAuthor = exports.DxAlert = exports.DxChat = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const chat_1 = __importDefault(require("devextreme/ui/chat"));
const index_2 = require("./core/index");
const componentConfig = {
    props: {
        accessKey: String,
        activeStateEnabled: Boolean,
        alerts: Array,
        dataSource: [Array, Object, String],
        dayHeaderFormat: [Object, String, Function],
        disabled: Boolean,
        elementAttr: Object,
        focusStateEnabled: Boolean,
        height: [Function, Number, String],
        hint: String,
        hoverStateEnabled: Boolean,
        items: Array,
        messageTemplate: {},
        messageTimestampFormat: [Object, String, Function],
        onDisposing: Function,
        onInitialized: Function,
        onMessageEntered: Function,
        onOptionChanged: Function,
        onTypingEnd: Function,
        onTypingStart: Function,
        reloadOnChange: Boolean,
        rtlEnabled: Boolean,
        showAvatar: Boolean,
        showDayHeaders: Boolean,
        showMessageTimestamp: Boolean,
        showUserName: Boolean,
        typingUsers: Array,
        user: Object,
        visible: Boolean,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:accessKey": null,
        "update:activeStateEnabled": null,
        "update:alerts": null,
        "update:dataSource": null,
        "update:dayHeaderFormat": null,
        "update:disabled": null,
        "update:elementAttr": null,
        "update:focusStateEnabled": null,
        "update:height": null,
        "update:hint": null,
        "update:hoverStateEnabled": null,
        "update:items": null,
        "update:messageTemplate": null,
        "update:messageTimestampFormat": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onMessageEntered": null,
        "update:onOptionChanged": null,
        "update:onTypingEnd": null,
        "update:onTypingStart": null,
        "update:reloadOnChange": null,
        "update:rtlEnabled": null,
        "update:showAvatar": null,
        "update:showDayHeaders": null,
        "update:showMessageTimestamp": null,
        "update:showUserName": null,
        "update:typingUsers": null,
        "update:user": null,
        "update:visible": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = chat_1.default;
        this.$_hasAsyncTemplate = true;
        this.$_expectedChildren = {
            alert: { isCollectionItem: true, optionName: "alerts" },
            dayHeaderFormat: { isCollectionItem: false, optionName: "dayHeaderFormat" },
            item: { isCollectionItem: true, optionName: "items" },
            messageTimestampFormat: { isCollectionItem: false, optionName: "messageTimestampFormat" },
            typingUser: { isCollectionItem: true, optionName: "typingUsers" },
            user: { isCollectionItem: false, optionName: "user" }
        };
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxChat = (0, vue_1.defineComponent)(componentConfig);
exports.DxChat = DxChat;
const DxAlertConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:id": null,
        "update:message": null,
    },
    props: {
        id: [Number, String],
        message: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxAlertConfig);
const DxAlert = (0, vue_1.defineComponent)(DxAlertConfig);
exports.DxAlert = DxAlert;
DxAlert.$_optionName = "alerts";
DxAlert.$_isCollectionItem = true;
const DxAuthorConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:avatarAlt": null,
        "update:avatarUrl": null,
        "update:id": null,
        "update:name": null,
    },
    props: {
        avatarAlt: String,
        avatarUrl: String,
        id: [Number, String],
        name: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxAuthorConfig);
const DxAuthor = (0, vue_1.defineComponent)(DxAuthorConfig);
exports.DxAuthor = DxAuthor;
DxAuthor.$_optionName = "author";
const DxDayHeaderFormatConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:currency": null,
        "update:formatter": null,
        "update:parser": null,
        "update:precision": null,
        "update:type": null,
        "update:useCurrencyAccountingStyle": null,
    },
    props: {
        currency: String,
        formatter: Function,
        parser: Function,
        precision: Number,
        type: String,
        useCurrencyAccountingStyle: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxDayHeaderFormatConfig);
const DxDayHeaderFormat = (0, vue_1.defineComponent)(DxDayHeaderFormatConfig);
exports.DxDayHeaderFormat = DxDayHeaderFormat;
DxDayHeaderFormat.$_optionName = "dayHeaderFormat";
const DxItemConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:author": null,
        "update:id": null,
        "update:text": null,
        "update:timestamp": null,
    },
    props: {
        author: Object,
        id: [Number, String],
        text: String,
        timestamp: [Date, Number, String]
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxItemConfig);
const DxItem = (0, vue_1.defineComponent)(DxItemConfig);
exports.DxItem = DxItem;
DxItem.$_optionName = "items";
DxItem.$_isCollectionItem = true;
DxItem.$_expectedChildren = {
    author: { isCollectionItem: false, optionName: "author" }
};
const DxMessageTimestampFormatConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:currency": null,
        "update:formatter": null,
        "update:parser": null,
        "update:precision": null,
        "update:type": null,
        "update:useCurrencyAccountingStyle": null,
    },
    props: {
        currency: String,
        formatter: Function,
        parser: Function,
        precision: Number,
        type: String,
        useCurrencyAccountingStyle: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxMessageTimestampFormatConfig);
const DxMessageTimestampFormat = (0, vue_1.defineComponent)(DxMessageTimestampFormatConfig);
exports.DxMessageTimestampFormat = DxMessageTimestampFormat;
DxMessageTimestampFormat.$_optionName = "messageTimestampFormat";
const DxTypingUserConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:avatarAlt": null,
        "update:avatarUrl": null,
        "update:id": null,
        "update:name": null,
    },
    props: {
        avatarAlt: String,
        avatarUrl: String,
        id: [Number, String],
        name: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxTypingUserConfig);
const DxTypingUser = (0, vue_1.defineComponent)(DxTypingUserConfig);
exports.DxTypingUser = DxTypingUser;
DxTypingUser.$_optionName = "typingUsers";
DxTypingUser.$_isCollectionItem = true;
const DxUserConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:avatarAlt": null,
        "update:avatarUrl": null,
        "update:id": null,
        "update:name": null,
    },
    props: {
        avatarAlt: String,
        avatarUrl: String,
        id: [Number, String],
        name: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxUserConfig);
const DxUser = (0, vue_1.defineComponent)(DxUserConfig);
exports.DxUser = DxUser;
DxUser.$_optionName = "user";
exports.default = DxChat;
