// devextreme-vue/cjs/diagram.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxZoomLevel = exports.DxViewToolbar = exports.DxToolboxGroup = exports.DxToolbox = exports.DxTabGroup = exports.DxTab = exports.DxPropertiesPanel = exports.DxPageSizeItem = exports.DxPageSize = exports.DxNodes = exports.DxMainToolbar = exports.DxItem = exports.DxHistoryToolbar = exports.DxGroup = exports.DxGridSize = exports.DxExport = exports.DxEditing = exports.DxEdges = exports.DxDefaultItemProperties = exports.DxCustomShape = exports.DxContextToolbox = exports.DxContextMenu = exports.DxConnectionPoint = exports.DxCommandItem = exports.DxCommand = exports.DxAutoLayout = exports.DxDiagram = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const diagram_1 = __importDefault(require("devextreme/ui/diagram"));
const index_2 = require("./core/index");
const componentConfig = {
    props: {
        autoZoomMode: String,
        contextMenu: Object,
        contextToolbox: Object,
        customShapes: Array,
        customShapeTemplate: {},
        customShapeToolboxTemplate: {},
        defaultItemProperties: Object,
        disabled: Boolean,
        edges: Object,
        editing: Object,
        elementAttr: Object,
        export: Object,
        fullScreen: Boolean,
        gridSize: [Number, Object],
        hasChanges: Boolean,
        height: [Function, Number, String],
        historyToolbar: Object,
        mainToolbar: Object,
        nodes: Object,
        onContentReady: Function,
        onCustomCommand: Function,
        onDisposing: Function,
        onInitialized: Function,
        onItemClick: Function,
        onItemDblClick: Function,
        onOptionChanged: Function,
        onRequestEditOperation: Function,
        onRequestLayoutUpdate: Function,
        onSelectionChanged: Function,
        pageColor: String,
        pageOrientation: String,
        pageSize: Object,
        propertiesPanel: Object,
        readOnly: Boolean,
        rtlEnabled: Boolean,
        showGrid: Boolean,
        simpleView: Boolean,
        snapToGrid: Boolean,
        toolbox: Object,
        units: String,
        useNativeScrolling: Boolean,
        viewToolbar: Object,
        viewUnits: String,
        visible: Boolean,
        width: [Function, Number, String],
        zoomLevel: [Number, Object]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:autoZoomMode": null,
        "update:contextMenu": null,
        "update:contextToolbox": null,
        "update:customShapes": null,
        "update:customShapeTemplate": null,
        "update:customShapeToolboxTemplate": null,
        "update:defaultItemProperties": null,
        "update:disabled": null,
        "update:edges": null,
        "update:editing": null,
        "update:elementAttr": null,
        "update:export": null,
        "update:fullScreen": null,
        "update:gridSize": null,
        "update:hasChanges": null,
        "update:height": null,
        "update:historyToolbar": null,
        "update:mainToolbar": null,
        "update:nodes": null,
        "update:onContentReady": null,
        "update:onCustomCommand": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onItemClick": null,
        "update:onItemDblClick": null,
        "update:onOptionChanged": null,
        "update:onRequestEditOperation": null,
        "update:onRequestLayoutUpdate": null,
        "update:onSelectionChanged": null,
        "update:pageColor": null,
        "update:pageOrientation": null,
        "update:pageSize": null,
        "update:propertiesPanel": null,
        "update:readOnly": null,
        "update:rtlEnabled": null,
        "update:showGrid": null,
        "update:simpleView": null,
        "update:snapToGrid": null,
        "update:toolbox": null,
        "update:units": null,
        "update:useNativeScrolling": null,
        "update:viewToolbar": null,
        "update:viewUnits": null,
        "update:visible": null,
        "update:width": null,
        "update:zoomLevel": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = diagram_1.default;
        this.$_hasAsyncTemplate = true;
        this.$_expectedChildren = {
            contextMenu: { isCollectionItem: false, optionName: "contextMenu" },
            contextToolbox: { isCollectionItem: false, optionName: "contextToolbox" },
            customShape: { isCollectionItem: true, optionName: "customShapes" },
            defaultItemProperties: { isCollectionItem: false, optionName: "defaultItemProperties" },
            edges: { isCollectionItem: false, optionName: "edges" },
            editing: { isCollectionItem: false, optionName: "editing" },
            export: { isCollectionItem: false, optionName: "export" },
            gridSize: { isCollectionItem: false, optionName: "gridSize" },
            historyToolbar: { isCollectionItem: false, optionName: "historyToolbar" },
            mainToolbar: { isCollectionItem: false, optionName: "mainToolbar" },
            nodes: { isCollectionItem: false, optionName: "nodes" },
            pageSize: { isCollectionItem: false, optionName: "pageSize" },
            propertiesPanel: { isCollectionItem: false, optionName: "propertiesPanel" },
            toolbox: { isCollectionItem: false, optionName: "toolbox" },
            viewToolbar: { isCollectionItem: false, optionName: "viewToolbar" },
            zoomLevel: { isCollectionItem: false, optionName: "zoomLevel" }
        };
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxDiagram = (0, vue_1.defineComponent)(componentConfig);
exports.DxDiagram = DxDiagram;
const DxAutoLayoutConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:orientation": null,
        "update:type": null,
    },
    props: {
        orientation: String,
        type: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxAutoLayoutConfig);
const DxAutoLayout = (0, vue_1.defineComponent)(DxAutoLayoutConfig);
exports.DxAutoLayout = DxAutoLayout;
DxAutoLayout.$_optionName = "autoLayout";
const DxCommandConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:icon": null,
        "update:items": null,
        "update:location": null,
        "update:name": null,
        "update:text": null,
    },
    props: {
        icon: String,
        items: Array,
        location: String,
        name: String,
        text: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxCommandConfig);
const DxCommand = (0, vue_1.defineComponent)(DxCommandConfig);
exports.DxCommand = DxCommand;
DxCommand.$_optionName = "commands";
DxCommand.$_isCollectionItem = true;
DxCommand.$_expectedChildren = {
    item: { isCollectionItem: true, optionName: "items" }
};
const DxCommandItemConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:icon": null,
        "update:items": null,
        "update:location": null,
        "update:name": null,
        "update:text": null,
    },
    props: {
        icon: String,
        items: Array,
        location: String,
        name: String,
        text: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxCommandItemConfig);
const DxCommandItem = (0, vue_1.defineComponent)(DxCommandItemConfig);
exports.DxCommandItem = DxCommandItem;
DxCommandItem.$_optionName = "items";
DxCommandItem.$_isCollectionItem = true;
const DxConnectionPointConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:x": null,
        "update:y": null,
    },
    props: {
        x: Number,
        y: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxConnectionPointConfig);
const DxConnectionPoint = (0, vue_1.defineComponent)(DxConnectionPointConfig);
exports.DxConnectionPoint = DxConnectionPoint;
DxConnectionPoint.$_optionName = "connectionPoints";
DxConnectionPoint.$_isCollectionItem = true;
const DxContextMenuConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:commands": null,
        "update:enabled": null,
    },
    props: {
        commands: Array,
        enabled: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxContextMenuConfig);
const DxContextMenu = (0, vue_1.defineComponent)(DxContextMenuConfig);
exports.DxContextMenu = DxContextMenu;
DxContextMenu.$_optionName = "contextMenu";
DxContextMenu.$_expectedChildren = {
    command: { isCollectionItem: true, optionName: "commands" }
};
const DxContextToolboxConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:category": null,
        "update:displayMode": null,
        "update:enabled": null,
        "update:shapeIconsPerRow": null,
        "update:shapes": null,
        "update:width": null,
    },
    props: {
        category: String,
        displayMode: String,
        enabled: Boolean,
        shapeIconsPerRow: Number,
        shapes: Array,
        width: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxContextToolboxConfig);
const DxContextToolbox = (0, vue_1.defineComponent)(DxContextToolboxConfig);
exports.DxContextToolbox = DxContextToolbox;
DxContextToolbox.$_optionName = "contextToolbox";
const DxCustomShapeConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:allowEditImage": null,
        "update:allowEditText": null,
        "update:allowResize": null,
        "update:backgroundImageHeight": null,
        "update:backgroundImageLeft": null,
        "update:backgroundImageToolboxUrl": null,
        "update:backgroundImageTop": null,
        "update:backgroundImageUrl": null,
        "update:backgroundImageWidth": null,
        "update:baseType": null,
        "update:category": null,
        "update:connectionPoints": null,
        "update:defaultHeight": null,
        "update:defaultImageUrl": null,
        "update:defaultText": null,
        "update:defaultWidth": null,
        "update:imageHeight": null,
        "update:imageLeft": null,
        "update:imageTop": null,
        "update:imageWidth": null,
        "update:keepRatioOnAutoSize": null,
        "update:maxHeight": null,
        "update:maxWidth": null,
        "update:minHeight": null,
        "update:minWidth": null,
        "update:template": null,
        "update:templateHeight": null,
        "update:templateLeft": null,
        "update:templateTop": null,
        "update:templateWidth": null,
        "update:textHeight": null,
        "update:textLeft": null,
        "update:textTop": null,
        "update:textWidth": null,
        "update:title": null,
        "update:toolboxTemplate": null,
        "update:toolboxWidthToHeightRatio": null,
        "update:type": null,
    },
    props: {
        allowEditImage: Boolean,
        allowEditText: Boolean,
        allowResize: Boolean,
        backgroundImageHeight: Number,
        backgroundImageLeft: Number,
        backgroundImageToolboxUrl: String,
        backgroundImageTop: Number,
        backgroundImageUrl: String,
        backgroundImageWidth: Number,
        baseType: String,
        category: String,
        connectionPoints: Array,
        defaultHeight: Number,
        defaultImageUrl: String,
        defaultText: String,
        defaultWidth: Number,
        imageHeight: Number,
        imageLeft: Number,
        imageTop: Number,
        imageWidth: Number,
        keepRatioOnAutoSize: Boolean,
        maxHeight: Number,
        maxWidth: Number,
        minHeight: Number,
        minWidth: Number,
        template: {},
        templateHeight: Number,
        templateLeft: Number,
        templateTop: Number,
        templateWidth: Number,
        textHeight: Number,
        textLeft: Number,
        textTop: Number,
        textWidth: Number,
        title: String,
        toolboxTemplate: {},
        toolboxWidthToHeightRatio: Number,
        type: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxCustomShapeConfig);
const DxCustomShape = (0, vue_1.defineComponent)(DxCustomShapeConfig);
exports.DxCustomShape = DxCustomShape;
DxCustomShape.$_optionName = "customShapes";
DxCustomShape.$_isCollectionItem = true;
DxCustomShape.$_expectedChildren = {
    connectionPoint: { isCollectionItem: true, optionName: "connectionPoints" }
};
const DxDefaultItemPropertiesConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:connectorLineEnd": null,
        "update:connectorLineStart": null,
        "update:connectorLineType": null,
        "update:shapeMaxHeight": null,
        "update:shapeMaxWidth": null,
        "update:shapeMinHeight": null,
        "update:shapeMinWidth": null,
        "update:style": null,
        "update:textStyle": null,
    },
    props: {
        connectorLineEnd: String,
        connectorLineStart: String,
        connectorLineType: String,
        shapeMaxHeight: Number,
        shapeMaxWidth: Number,
        shapeMinHeight: Number,
        shapeMinWidth: Number,
        style: Object,
        textStyle: Object
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxDefaultItemPropertiesConfig);
const DxDefaultItemProperties = (0, vue_1.defineComponent)(DxDefaultItemPropertiesConfig);
exports.DxDefaultItemProperties = DxDefaultItemProperties;
DxDefaultItemProperties.$_optionName = "defaultItemProperties";
const DxEdgesConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:customDataExpr": null,
        "update:dataSource": null,
        "update:fromExpr": null,
        "update:fromLineEndExpr": null,
        "update:fromPointIndexExpr": null,
        "update:keyExpr": null,
        "update:lineTypeExpr": null,
        "update:lockedExpr": null,
        "update:pointsExpr": null,
        "update:styleExpr": null,
        "update:textExpr": null,
        "update:textStyleExpr": null,
        "update:toExpr": null,
        "update:toLineEndExpr": null,
        "update:toPointIndexExpr": null,
        "update:zIndexExpr": null,
    },
    props: {
        customDataExpr: [Function, String],
        dataSource: [Array, Object, String],
        fromExpr: [Function, String],
        fromLineEndExpr: [Function, String],
        fromPointIndexExpr: [Function, String],
        keyExpr: [Function, String],
        lineTypeExpr: [Function, String],
        lockedExpr: [Function, String],
        pointsExpr: [Function, String],
        styleExpr: [Function, String],
        textExpr: [Function, String],
        textStyleExpr: [Function, String],
        toExpr: [Function, String],
        toLineEndExpr: [Function, String],
        toPointIndexExpr: [Function, String],
        zIndexExpr: [Function, String]
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxEdgesConfig);
const DxEdges = (0, vue_1.defineComponent)(DxEdgesConfig);
exports.DxEdges = DxEdges;
DxEdges.$_optionName = "edges";
const DxEditingConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:allowAddShape": null,
        "update:allowChangeConnection": null,
        "update:allowChangeConnectorPoints": null,
        "update:allowChangeConnectorText": null,
        "update:allowChangeShapeText": null,
        "update:allowDeleteConnector": null,
        "update:allowDeleteShape": null,
        "update:allowMoveShape": null,
        "update:allowResizeShape": null,
    },
    props: {
        allowAddShape: Boolean,
        allowChangeConnection: Boolean,
        allowChangeConnectorPoints: Boolean,
        allowChangeConnectorText: Boolean,
        allowChangeShapeText: Boolean,
        allowDeleteConnector: Boolean,
        allowDeleteShape: Boolean,
        allowMoveShape: Boolean,
        allowResizeShape: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxEditingConfig);
const DxEditing = (0, vue_1.defineComponent)(DxEditingConfig);
exports.DxEditing = DxEditing;
DxEditing.$_optionName = "editing";
const DxExportConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:fileName": null,
    },
    props: {
        fileName: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxExportConfig);
const DxExport = (0, vue_1.defineComponent)(DxExportConfig);
exports.DxExport = DxExport;
DxExport.$_optionName = "export";
const DxGridSizeConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:items": null,
        "update:value": null,
    },
    props: {
        items: Array,
        value: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxGridSizeConfig);
const DxGridSize = (0, vue_1.defineComponent)(DxGridSizeConfig);
exports.DxGridSize = DxGridSize;
DxGridSize.$_optionName = "gridSize";
const DxGroupConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:category": null,
        "update:commands": null,
        "update:displayMode": null,
        "update:expanded": null,
        "update:shapes": null,
        "update:title": null,
    },
    props: {
        category: String,
        commands: Array,
        displayMode: String,
        expanded: Boolean,
        shapes: Array,
        title: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxGroupConfig);
const DxGroup = (0, vue_1.defineComponent)(DxGroupConfig);
exports.DxGroup = DxGroup;
DxGroup.$_optionName = "groups";
DxGroup.$_isCollectionItem = true;
DxGroup.$_expectedChildren = {
    command: { isCollectionItem: true, optionName: "commands" }
};
const DxHistoryToolbarConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:commands": null,
        "update:visible": null,
    },
    props: {
        commands: Array,
        visible: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxHistoryToolbarConfig);
const DxHistoryToolbar = (0, vue_1.defineComponent)(DxHistoryToolbarConfig);
exports.DxHistoryToolbar = DxHistoryToolbar;
DxHistoryToolbar.$_optionName = "historyToolbar";
DxHistoryToolbar.$_expectedChildren = {
    command: { isCollectionItem: true, optionName: "commands" }
};
const DxItemConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:height": null,
        "update:icon": null,
        "update:items": null,
        "update:location": null,
        "update:name": null,
        "update:text": null,
        "update:width": null,
    },
    props: {
        height: Number,
        icon: String,
        items: Array,
        location: String,
        name: String,
        text: String,
        width: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxItemConfig);
const DxItem = (0, vue_1.defineComponent)(DxItemConfig);
exports.DxItem = DxItem;
DxItem.$_optionName = "items";
DxItem.$_isCollectionItem = true;
const DxMainToolbarConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:commands": null,
        "update:visible": null,
    },
    props: {
        commands: Array,
        visible: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxMainToolbarConfig);
const DxMainToolbar = (0, vue_1.defineComponent)(DxMainToolbarConfig);
exports.DxMainToolbar = DxMainToolbar;
DxMainToolbar.$_optionName = "mainToolbar";
DxMainToolbar.$_expectedChildren = {
    command: { isCollectionItem: true, optionName: "commands" }
};
const DxNodesConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:autoLayout": null,
        "update:autoSizeEnabled": null,
        "update:containerChildrenExpr": null,
        "update:containerKeyExpr": null,
        "update:customDataExpr": null,
        "update:dataSource": null,
        "update:heightExpr": null,
        "update:imageUrlExpr": null,
        "update:itemsExpr": null,
        "update:keyExpr": null,
        "update:leftExpr": null,
        "update:lockedExpr": null,
        "update:parentKeyExpr": null,
        "update:styleExpr": null,
        "update:textExpr": null,
        "update:textStyleExpr": null,
        "update:topExpr": null,
        "update:typeExpr": null,
        "update:widthExpr": null,
        "update:zIndexExpr": null,
    },
    props: {
        autoLayout: [String, Object],
        autoSizeEnabled: Boolean,
        containerChildrenExpr: [Function, String],
        containerKeyExpr: [Function, String],
        customDataExpr: [Function, String],
        dataSource: [Array, Object, String],
        heightExpr: [Function, String],
        imageUrlExpr: [Function, String],
        itemsExpr: [Function, String],
        keyExpr: [Function, String],
        leftExpr: [Function, String],
        lockedExpr: [Function, String],
        parentKeyExpr: [Function, String],
        styleExpr: [Function, String],
        textExpr: [Function, String],
        textStyleExpr: [Function, String],
        topExpr: [Function, String],
        typeExpr: [Function, String],
        widthExpr: [Function, String],
        zIndexExpr: [Function, String]
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxNodesConfig);
const DxNodes = (0, vue_1.defineComponent)(DxNodesConfig);
exports.DxNodes = DxNodes;
DxNodes.$_optionName = "nodes";
DxNodes.$_expectedChildren = {
    autoLayout: { isCollectionItem: false, optionName: "autoLayout" }
};
const DxPageSizeConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:height": null,
        "update:items": null,
        "update:width": null,
    },
    props: {
        height: Number,
        items: Array,
        width: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxPageSizeConfig);
const DxPageSize = (0, vue_1.defineComponent)(DxPageSizeConfig);
exports.DxPageSize = DxPageSize;
DxPageSize.$_optionName = "pageSize";
DxPageSize.$_expectedChildren = {
    item: { isCollectionItem: true, optionName: "items" },
    pageSizeItem: { isCollectionItem: true, optionName: "items" }
};
const DxPageSizeItemConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:height": null,
        "update:text": null,
        "update:width": null,
    },
    props: {
        height: Number,
        text: String,
        width: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxPageSizeItemConfig);
const DxPageSizeItem = (0, vue_1.defineComponent)(DxPageSizeItemConfig);
exports.DxPageSizeItem = DxPageSizeItem;
DxPageSizeItem.$_optionName = "items";
DxPageSizeItem.$_isCollectionItem = true;
const DxPropertiesPanelConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:tabs": null,
        "update:visibility": null,
    },
    props: {
        tabs: Array,
        visibility: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxPropertiesPanelConfig);
const DxPropertiesPanel = (0, vue_1.defineComponent)(DxPropertiesPanelConfig);
exports.DxPropertiesPanel = DxPropertiesPanel;
DxPropertiesPanel.$_optionName = "propertiesPanel";
DxPropertiesPanel.$_expectedChildren = {
    tab: { isCollectionItem: true, optionName: "tabs" }
};
const DxTabConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:commands": null,
        "update:groups": null,
        "update:title": null,
    },
    props: {
        commands: Array,
        groups: Array,
        title: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxTabConfig);
const DxTab = (0, vue_1.defineComponent)(DxTabConfig);
exports.DxTab = DxTab;
DxTab.$_optionName = "tabs";
DxTab.$_isCollectionItem = true;
DxTab.$_expectedChildren = {
    command: { isCollectionItem: true, optionName: "commands" },
    group: { isCollectionItem: true, optionName: "groups" },
    tabGroup: { isCollectionItem: true, optionName: "groups" }
};
const DxTabGroupConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:commands": null,
        "update:title": null,
    },
    props: {
        commands: Array,
        title: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxTabGroupConfig);
const DxTabGroup = (0, vue_1.defineComponent)(DxTabGroupConfig);
exports.DxTabGroup = DxTabGroup;
DxTabGroup.$_optionName = "groups";
DxTabGroup.$_isCollectionItem = true;
DxTabGroup.$_expectedChildren = {
    command: { isCollectionItem: true, optionName: "commands" }
};
const DxToolboxConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:groups": null,
        "update:shapeIconsPerRow": null,
        "update:showSearch": null,
        "update:visibility": null,
        "update:width": null,
    },
    props: {
        groups: Array,
        shapeIconsPerRow: Number,
        showSearch: Boolean,
        visibility: String,
        width: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxToolboxConfig);
const DxToolbox = (0, vue_1.defineComponent)(DxToolboxConfig);
exports.DxToolbox = DxToolbox;
DxToolbox.$_optionName = "toolbox";
DxToolbox.$_expectedChildren = {
    group: { isCollectionItem: true, optionName: "groups" },
    toolboxGroup: { isCollectionItem: true, optionName: "groups" }
};
const DxToolboxGroupConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:category": null,
        "update:displayMode": null,
        "update:expanded": null,
        "update:shapes": null,
        "update:title": null,
    },
    props: {
        category: String,
        displayMode: String,
        expanded: Boolean,
        shapes: Array,
        title: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxToolboxGroupConfig);
const DxToolboxGroup = (0, vue_1.defineComponent)(DxToolboxGroupConfig);
exports.DxToolboxGroup = DxToolboxGroup;
DxToolboxGroup.$_optionName = "groups";
DxToolboxGroup.$_isCollectionItem = true;
const DxViewToolbarConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:commands": null,
        "update:visible": null,
    },
    props: {
        commands: Array,
        visible: Boolean
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxViewToolbarConfig);
const DxViewToolbar = (0, vue_1.defineComponent)(DxViewToolbarConfig);
exports.DxViewToolbar = DxViewToolbar;
DxViewToolbar.$_optionName = "viewToolbar";
DxViewToolbar.$_expectedChildren = {
    command: { isCollectionItem: true, optionName: "commands" }
};
const DxZoomLevelConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:items": null,
        "update:value": null,
    },
    props: {
        items: Array,
        value: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxZoomLevelConfig);
const DxZoomLevel = (0, vue_1.defineComponent)(DxZoomLevelConfig);
exports.DxZoomLevel = DxZoomLevel;
DxZoomLevel.$_optionName = "zoomLevel";
exports.default = DxDiagram;
