// devextreme-vue/cjs/pivot-grid-field-chooser.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxTexts = exports.DxSearch = exports.DxPivotGridFieldChooserTexts = exports.DxHeaderFilterTexts = exports.DxHeaderFilter = exports.DxPivotGridFieldChooser = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const pivot_grid_field_chooser_1 = __importDefault(require("devextreme/ui/pivot_grid_field_chooser"));
const index_2 = require("./core/index");
const componentConfig = {
    props: {
        accessKey: String,
        activeStateEnabled: Boolean,
        allowSearch: Boolean,
        applyChangesMode: String,
        dataSource: Object,
        disabled: Boolean,
        elementAttr: Object,
        encodeHtml: Boolean,
        focusStateEnabled: Boolean,
        headerFilter: Object,
        height: [Function, Number, String],
        hint: String,
        hoverStateEnabled: Boolean,
        layout: Number,
        onContentReady: Function,
        onContextMenuPreparing: Function,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        rtlEnabled: Boolean,
        searchTimeout: Number,
        state: {},
        tabIndex: Number,
        texts: Object,
        visible: Boolean,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:accessKey": null,
        "update:activeStateEnabled": null,
        "update:allowSearch": null,
        "update:applyChangesMode": null,
        "update:dataSource": null,
        "update:disabled": null,
        "update:elementAttr": null,
        "update:encodeHtml": null,
        "update:focusStateEnabled": null,
        "update:headerFilter": null,
        "update:height": null,
        "update:hint": null,
        "update:hoverStateEnabled": null,
        "update:layout": null,
        "update:onContentReady": null,
        "update:onContextMenuPreparing": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:rtlEnabled": null,
        "update:searchTimeout": null,
        "update:state": null,
        "update:tabIndex": null,
        "update:texts": null,
        "update:visible": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = pivot_grid_field_chooser_1.default;
        this.$_hasAsyncTemplate = true;
        this.$_expectedChildren = {
            headerFilter: { isCollectionItem: false, optionName: "headerFilter" },
            pivotGridFieldChooserTexts: { isCollectionItem: false, optionName: "texts" },
            texts: { isCollectionItem: false, optionName: "texts" }
        };
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxPivotGridFieldChooser = (0, vue_1.defineComponent)(componentConfig);
exports.DxPivotGridFieldChooser = DxPivotGridFieldChooser;
const DxHeaderFilterConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:allowSearch": null,
        "update:allowSelectAll": null,
        "update:height": null,
        "update:search": null,
        "update:searchTimeout": null,
        "update:showRelevantValues": null,
        "update:texts": null,
        "update:width": null,
    },
    props: {
        allowSearch: Boolean,
        allowSelectAll: Boolean,
        height: Number,
        search: Object,
        searchTimeout: Number,
        showRelevantValues: Boolean,
        texts: Object,
        width: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxHeaderFilterConfig);
const DxHeaderFilter = (0, vue_1.defineComponent)(DxHeaderFilterConfig);
exports.DxHeaderFilter = DxHeaderFilter;
DxHeaderFilter.$_optionName = "headerFilter";
DxHeaderFilter.$_expectedChildren = {
    headerFilterTexts: { isCollectionItem: false, optionName: "texts" },
    search: { isCollectionItem: false, optionName: "search" },
    texts: { isCollectionItem: false, optionName: "texts" }
};
const DxHeaderFilterTextsConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:cancel": null,
        "update:emptyValue": null,
        "update:ok": null,
    },
    props: {
        cancel: String,
        emptyValue: String,
        ok: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxHeaderFilterTextsConfig);
const DxHeaderFilterTexts = (0, vue_1.defineComponent)(DxHeaderFilterTextsConfig);
exports.DxHeaderFilterTexts = DxHeaderFilterTexts;
DxHeaderFilterTexts.$_optionName = "texts";
const DxPivotGridFieldChooserTextsConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:allFields": null,
        "update:columnFields": null,
        "update:dataFields": null,
        "update:filterFields": null,
        "update:rowFields": null,
    },
    props: {
        allFields: String,
        columnFields: String,
        dataFields: String,
        filterFields: String,
        rowFields: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxPivotGridFieldChooserTextsConfig);
const DxPivotGridFieldChooserTexts = (0, vue_1.defineComponent)(DxPivotGridFieldChooserTextsConfig);
exports.DxPivotGridFieldChooserTexts = DxPivotGridFieldChooserTexts;
DxPivotGridFieldChooserTexts.$_optionName = "texts";
const DxSearchConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:editorOptions": null,
        "update:enabled": null,
        "update:mode": null,
        "update:timeout": null,
    },
    props: {
        editorOptions: {},
        enabled: Boolean,
        mode: String,
        timeout: Number
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxSearchConfig);
const DxSearch = (0, vue_1.defineComponent)(DxSearchConfig);
exports.DxSearch = DxSearch;
DxSearch.$_optionName = "search";
const DxTextsConfig = {
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:allFields": null,
        "update:cancel": null,
        "update:columnFields": null,
        "update:dataFields": null,
        "update:emptyValue": null,
        "update:filterFields": null,
        "update:ok": null,
        "update:rowFields": null,
    },
    props: {
        allFields: String,
        cancel: String,
        columnFields: String,
        dataFields: String,
        emptyValue: String,
        filterFields: String,
        ok: String,
        rowFields: String
    }
};
(0, index_2.prepareConfigurationComponentConfig)(DxTextsConfig);
const DxTexts = (0, vue_1.defineComponent)(DxTextsConfig);
exports.DxTexts = DxTexts;
DxTexts.$_optionName = "texts";
exports.default = DxPivotGridFieldChooser;
