// devextreme-vue/cjs/resizable.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxResizable = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const resizable_1 = __importDefault(require("devextreme/ui/resizable"));
const componentConfig = {
    props: {
        area: {},
        elementAttr: Object,
        handles: String,
        height: [Function, Number, String],
        keepAspectRatio: Boolean,
        maxHeight: Number,
        maxWidth: Number,
        minHeight: Number,
        minWidth: Number,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        onResize: Function,
        onResizeEnd: Function,
        onResizeStart: Function,
        rtlEnabled: Boolean,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:area": null,
        "update:elementAttr": null,
        "update:handles": null,
        "update:height": null,
        "update:keepAspectRatio": null,
        "update:maxHeight": null,
        "update:maxWidth": null,
        "update:minHeight": null,
        "update:minWidth": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:onResize": null,
        "update:onResizeEnd": null,
        "update:onResizeStart": null,
        "update:rtlEnabled": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = resizable_1.default;
        this.$_hasAsyncTemplate = true;
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxResizable = (0, vue_1.defineComponent)(componentConfig);
exports.DxResizable = DxResizable;
exports.default = DxResizable;
