// devextreme-vue/cjs/speed-dial-action.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxSpeedDialAction = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const speed_dial_action_1 = __importDefault(require("devextreme/ui/speed_dial_action"));
const componentConfig = {
    props: {
        accessKey: String,
        activeStateEnabled: Boolean,
        elementAttr: Object,
        focusStateEnabled: Boolean,
        hint: String,
        hoverStateEnabled: Boolean,
        icon: String,
        index: Number,
        label: String,
        onClick: Function,
        onContentReady: Function,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        rtlEnabled: Boolean,
        tabIndex: Number,
        visible: Boolean
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:accessKey": null,
        "update:activeStateEnabled": null,
        "update:elementAttr": null,
        "update:focusStateEnabled": null,
        "update:hint": null,
        "update:hoverStateEnabled": null,
        "update:icon": null,
        "update:index": null,
        "update:label": null,
        "update:onClick": null,
        "update:onContentReady": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:rtlEnabled": null,
        "update:tabIndex": null,
        "update:visible": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = speed_dial_action_1.default;
        this.$_hasAsyncTemplate = true;
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxSpeedDialAction = (0, vue_1.defineComponent)(componentConfig);
exports.DxSpeedDialAction = DxSpeedDialAction;
exports.default = DxSpeedDialAction;
