// devextreme-vue/cjs/validation-group.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DxValidationGroup = void 0;
const vue_1 = require("vue");
const index_1 = require("./core/index");
const validation_group_1 = __importDefault(require("devextreme/ui/validation_group"));
const componentConfig = {
    props: {
        elementAttr: Object,
        height: [Function, Number, String],
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:elementAttr": null,
        "update:height": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = validation_group_1.default;
        this.$_hasAsyncTemplate = true;
    }
};
(0, index_1.prepareComponentConfig)(componentConfig);
const DxValidationGroup = (0, vue_1.defineComponent)(componentConfig);
exports.DxValidationGroup = DxValidationGroup;
exports.default = DxValidationGroup;
