// devextreme-vue@24.2.10/esm/pagination.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

import { defineComponent } from "vue";
import { prepareComponentConfig } from "./core/index";
import Pagination from "devextreme/ui/pagination";
const componentConfig = {
    props: {
        accessKey: String,
        activeStateEnabled: Boolean,
        allowedPageSizes: Array,
        disabled: Boolean,
        displayMode: String,
        elementAttr: Object,
        focusStateEnabled: Boolean,
        height: [Function, Number, String],
        hint: String,
        hoverStateEnabled: Boolean,
        infoText: String,
        itemCount: Number,
        label: String,
        onContentReady: Function,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        pageIndex: Number,
        pageSize: Number,
        rtlEnabled: Boolean,
        showInfo: Boolean,
        showNavigationButtons: Boolean,
        showPageSizeSelector: Boolean,
        tabIndex: Number,
        visible: Boolean,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:accessKey": null,
        "update:activeStateEnabled": null,
        "update:allowedPageSizes": null,
        "update:disabled": null,
        "update:displayMode": null,
        "update:elementAttr": null,
        "update:focusStateEnabled": null,
        "update:height": null,
        "update:hint": null,
        "update:hoverStateEnabled": null,
        "update:infoText": null,
        "update:itemCount": null,
        "update:label": null,
        "update:onContentReady": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:pageIndex": null,
        "update:pageSize": null,
        "update:rtlEnabled": null,
        "update:showInfo": null,
        "update:showNavigationButtons": null,
        "update:showPageSizeSelector": null,
        "update:tabIndex": null,
        "update:visible": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = Pagination;
        this.$_hasAsyncTemplate = true;
    }
};
prepareComponentConfig(componentConfig);
const DxPagination = defineComponent(componentConfig);
export default DxPagination;
export { DxPagination };
