// devextreme-vue@24.2.10/esm/resizable.js
/*!
 * devextreme-vue
 * Version: 24.2.10
 * Build date: Fri Aug 29 2025
 *
 * Copyright (c) 2012 - 2025 Developer Express Inc. ALL RIGHTS RESERVED
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file in the root of the project for details.
 *
 * https://github.com/DevExpress/devextreme-vue
 */

import { defineComponent } from "vue";
import { prepareComponentConfig } from "./core/index";
import Resizable from "devextreme/ui/resizable";
const componentConfig = {
    props: {
        area: {},
        elementAttr: Object,
        handles: String,
        height: [Function, Number, String],
        keepAspectRatio: Boolean,
        maxHeight: Number,
        maxWidth: Number,
        minHeight: Number,
        minWidth: Number,
        onDisposing: Function,
        onInitialized: Function,
        onOptionChanged: Function,
        onResize: Function,
        onResizeEnd: Function,
        onResizeStart: Function,
        rtlEnabled: Boolean,
        width: [Function, Number, String]
    },
    emits: {
        "update:isActive": null,
        "update:hoveredElement": null,
        "update:area": null,
        "update:elementAttr": null,
        "update:handles": null,
        "update:height": null,
        "update:keepAspectRatio": null,
        "update:maxHeight": null,
        "update:maxWidth": null,
        "update:minHeight": null,
        "update:minWidth": null,
        "update:onDisposing": null,
        "update:onInitialized": null,
        "update:onOptionChanged": null,
        "update:onResize": null,
        "update:onResizeEnd": null,
        "update:onResizeStart": null,
        "update:rtlEnabled": null,
        "update:width": null,
    },
    computed: {
        instance() {
            return this.$_instance;
        }
    },
    beforeCreate() {
        this.$_WidgetClass = Resizable;
        this.$_hasAsyncTemplate = true;
    }
};
prepareComponentConfig(componentConfig);
const DxResizable = defineComponent(componentConfig);
export default DxResizable;
export { DxResizable };
